src/Transport/Sending/DefaultBatchedOperationsToBrokeredMessagesConverter.cs

```diff
@@ -81,8 +80,16 @@ void SetReplyToAddress(OutgoingMessage outgoingMessage, BrokeredMessage brokered
 
                     if (selected != null)
                     {
-                        replyTo = mapper.Map(new EntityAddress(replyTo.Name, selected.Name));
+                        if (mapper is DefaultNamespaceNameToConnectionStringMapper)
```

Comment:
Checking against types is a smell. I don't like it :)

Anyway...the logic here isn't clear for me. I try to summarize what transport has to accomplish:
- it extracts `replyTo` header value
- if address doesn't contain a suffix (`@something`), transport adds ones
- transport maps built value using configured `mapper`

Is it right? If true, could we change the code as following?

``` csharp
if (outgoingMessage.Headers.ContainsKey(Headers.ReplyToAddress))
{
    var replyTo = new EntityAddress(outgoingMessage.Headers[Headers.ReplyToAddress]);
    if (!replyTo.HasSuffix)
    {
        var namespaces = settings.Get<NamespaceConfigurations(WellKnownConfigurationKeys.Topology.Addressing.Namespaces);
        var defaultName = settings.Get<string(WellKnownConfigurationKeys.Topology.Addressing.DefaultNamespaceName);
        var selected = namespaces.FirstOrDefault(ns => ns.Name == defaultName || 
                                                       ns.Purpose == NamespacePurpose.Partitioning);
        if (selected != null)
            replyTo = new EntityAddress(replyTo, selected.Name);
    }

    var convertedValue = mapper.Map(replyTo);
    outgoingMessage.Headers[Headers.ReplyToAddress] = convertedValue;
    brokeredMessage.ReplyTo = convertedValue;
}
```